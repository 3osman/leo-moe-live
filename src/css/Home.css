.Home {
  text-align: center;
  background: linear-gradient(141deg, #2cb5e8 0%, #1fc8db 51%, #7dfff7 75%);
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 60px;
  margin-right: 10px;
  margin-top:7px;
}

.hero {
  font-size: large;
  color: white;
}

.hero h3 {
  font-family: 'Dosis', sans-serif;
  opacity: 0.95;
}

.search-container {
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.logo-wrapper {
  display: flex !important;
  flex-direction: row !important;
  justify-content: center !important;
  align-items: center !important;
}

footer {
  font-size: large;
  position: relative;
  padding-top: 40px;
  width: 100%;
}

footer a:first-child {
  padding-right: 40px;
}
/*@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}*/
