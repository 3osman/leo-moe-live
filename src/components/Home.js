import React, { Component } from 'react'
import logo from './../logo_3.gif'
import Streamchoice from './Streamchoice'
import Searchbar from './Searchbar'
import {Row, Col} from 'react-materialize'
import {platforms} from './../utils/config'
import './../css/Home.css'

class Home extends Component {
  onChildChanged(type, value) {
    this.refs.searchbar.setSearchService(type, value)
  }
  render() {
    return (
      <div className="Home">
        <div className="search-container container">
          <Row className="hero">
            <h3 className="logo-wrapper">
              <img src={logo} className="App-logo" alt="logo" />
              Reemote
            </h3>
            <p>
              Search live videos around the internet
            </p>
          </Row>
          <Row>
            <Col s={12} m={12} l={12}>
              <Searchbar ref="searchbar" youtube="true" twitch="true" periscope="true">
              </Searchbar>
            </Col>
          </Row>
          <Row>
            {platforms.map(function(p, index){
                return (
                    <Col s={12} m={4} l={4} key={index}>
                      <Streamchoice
                        mini={false}
                        platform={p}
                        initialChecked="true"
                        callbackParent={(type, value) => this.onChildChanged(type, value)} />
                    </Col>
                )
              })
            }
          </Row>
          {/*<Row>
            <footer>
              <Link to="about" className="white-text">ABOUT</Link>
              <Link to="contacts" className="white-text">CONTACTS</Link>
            </footer>
          </Row>*/}
        </div>
      </div>
    )
  }
}

export default Home
