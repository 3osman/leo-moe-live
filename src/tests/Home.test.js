import React from 'react'
import ReactDOM from 'react-dom'
import { mount } from 'enzyme'
import Home from './../components/Home'
import renderer from 'react-test-renderer';

it('it renders correctly', () => {
  var homeRendered = renderer.create(
    <Home />
  )
  expect(homeRendered.toJSON()).toMatchSnapshot()
})
