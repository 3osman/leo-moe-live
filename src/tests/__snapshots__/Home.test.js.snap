exports[`test it renders correctly 1`] = `
<div
  className="Home">
  <div
    className="search-container container">
    <div
      className="row hero">
      <h3
        className="logo-wrapper">
        <img
          alt="logo"
          className="App-logo"
          src="logo_3.gif" />
        Reemote
      </h3>
      <p>
        Search live videos around the internet
      </p>
    </div>
    <div
      className="row">
      <div
        className="col s12 m12 l12">
        <div
          className="card-panel Searchbar">
          <a
            onClick={[Function]}
            style={Object {}}>
            <button
              className="btn disabled waves-effect waves-light btn-large btn-floating red lighten-2 search-button"
              icon="search"
              onClick={undefined}>
              <i
                className="material-icons">
                search
              </i>
            </button>
          </a>
          <input
            className="search-text grey-text grey-darken-4"
            name="search"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Search live stream.."
            type="text"
            value="" />
        </div>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col s12 m4 l4">
        <div
          className="card-panel">
          <input
            className="filled-in"
            defaultChecked="true"
            id="youtube-filled-in-box"
            onChange={[Function]}
            type="checkbox" />
          <label
            className="black-text"
            htmlFor="youtube-filled-in-box">
            <img
              alt="Logo"
              src="youtube-logo.svg" />
            Youtube Live
          </label>
        </div>
      </div>
      <div
        className="col s12 m4 l4">
        <div
          className="card-panel">
          <input
            className="filled-in"
            defaultChecked="true"
            id="twitch-filled-in-box"
            onChange={[Function]}
            type="checkbox" />
          <label
            className="black-text"
            htmlFor="twitch-filled-in-box">
            <img
              alt="Logo"
              src="twitch-logo.svg" />
            Twitch
          </label>
        </div>
      </div>
      <div
        className="col s12 m4 l4">
        <div
          className="card-panel">
          <input
            className="filled-in"
            defaultChecked="true"
            id="periscope-filled-in-box"
            onChange={[Function]}
            type="checkbox" />
          <label
            className="black-text"
            htmlFor="periscope-filled-in-box">
            <img
              alt="Logo"
              src="periscope-logo.svg" />
            Periscope
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
